File: ios/GhostChat/Models/Room.swift
import Foundation

struct Room: Codable, Equatable, Identifiable {
    let id: String
    let createdAt: Date
    let myRole: Role

    init(id: String, createdAt: Date = Date(), myRole: Role) {
        self.id = id
        self.createdAt = createdAt
        self.myRole = myRole
    }

    /// base64url, 64 chars → 48 raw bytes → 384 bits of entropy
    static func isValidID(_ candidate: String) -> Bool {
        guard candidate.count == 64 else { return false }
        return candidate.allSatisfy { ch in
            ch.isLetter || ch.isNumber || ch == "-" || ch == "_"
        }
    }
}

File: android/app/src/main/java/com/kordar/ghostchat/models/Room.kt
package com.kordar.ghostchat.models

import java.util.UUID

/**
 * Ephemeral session room. IDs are 48 random bytes → base64url (64 chars / 384-bit entropy).
 */
data class Room(
    val id: String,
    val createdAt: Long = System.currentTimeMillis(),
    val myRole: Role
) {
    companion object {
        /** Mirrors iOS `Room.isValidID`. */
        fun isValidId(candidate: String): Boolean {
            if (candidate.length != 64) return false
            return candidate.all { ch ->
                ch.isLetterOrDigit() || ch == '-' || ch == '_'
            }
        }

        /** Generate a fresh UUID-backed ID for tests only — production IDs come from the server. */
        internal fun randomTestingId(): String = UUID.randomUUID().toString().replace("-", "").padEnd(64, '_').substring(0, 64)
    }
}
